File: migrate-styles.js
#!/usr/bin/env node

/**
 * StyleSheet to TailwindCSS Migration Script
 * Migrates all StyleSheet.create() usage to TailwindCSS className in shared screens
 */

const fs = require('fs');
const path = require('path');

// Style property to Tailwind class mapping
const STYLE_MAP = {
  // Layout
  'flex: 1': 'flex-1',
  'flexDirection: \'row\'': 'flex-row',
  'flexDirection: \'column\'': 'flex-col',
  'flexDirection: \'row-reverse\'': 'flex-row-reverse',
  'justifyContent: \'center\'': 'justify-center',
  'justifyContent: \'space-between\'': 'justify-between',
  'justifyContent: \'flex-start\'': 'justify-start',
  'alignItems: \'center\'': 'items-center',
  'alignItems: \'flex-start\'': 'items-start',

  // Background colors (from colors theme)
  'backgroundColor: colors.background': 'bg-black',
  'backgroundColor: colors.backgroundLight': 'bg-white/5',
  'backgroundColor: colors.backgroundLighter': 'bg-white/10',
  'backgroundColor: colors.primary': 'bg-purple-600',
  'backgroundColor: \'rgba(0, 0, 0, 0.4)\'': 'bg-black/40',
  'backgroundColor: \'rgba(0, 0, 0, 0.5)\'': 'bg-black/50',
  'backgroundColor: \'rgba(255, 255, 255, 0.05)\'': 'bg-white/5',
  'backgroundColor: \'rgba(255, 255, 255, 0.1)\'': 'bg-white/10',
  'backgroundColor: \'rgba(168, 85, 247, 0.2)\'': 'bg-purple-500/20',

  // Text colors
  'color: colors.text': 'text-white',
  'color: colors.textSecondary': 'text-gray-300',
  'color: colors.textMuted': 'text-gray-500',
  'color: colors.primary': 'text-purple-600',
  'color: colors.error': 'text-red-500',

  // Spacing (using spacing theme values)
  'padding: spacing.sm': 'p-2',
  'padding: spacing.md': 'p-4',
  'padding: spacing.lg': 'p-6',
  'padding: spacing.xl': 'p-8',
  'padding: spacing.xxl': 'p-12',
  'margin: spacing.sm': 'm-2',
  'margin: spacing.md': 'm-4',
  'margin: spacing.lg': 'm-6',
  'marginTop: spacing.md': 'mt-4',
  'marginBottom: spacing.lg': 'mb-6',

  // Border radius
  'borderRadius: borderRadius.md': 'rounded-md',
  'borderRadius: borderRadius.lg': 'rounded-lg',
  'borderRadius: borderRadius.xl': 'rounded-xl',
  'borderRadius: borderRadius.full': 'rounded-full',

  // Typography
  'fontSize: 14': 'text-sm',
  'fontSize: 16': 'text-base',
  'fontSize: 18': 'text-lg',
  'fontSize: 20': 'text-xl',
  'fontSize: 24': 'text-2xl',
  'fontSize: 28': 'text-3xl',
  'fontWeight: \'600\'': 'font-semibold',
  'fontWeight: \'bold\'': 'font-bold',
  'textAlign: \'center\'': 'text-center',

  // Borders
  'borderWidth: 2': 'border-2',
  'borderColor: \'transparent\'': 'border-transparent',
  'borderColor: colors.primary': 'border-purple-600',

  // Overflow
  'overflow: \'hidden\'': 'overflow-hidden',

  // Position
  'position: \'absolute\'': 'absolute',
  'position: \'relative\'': 'relative',
};

function convertStyleObjectToTailwind(styleObj) {
  // This is a simplified converter - for complex styles, manual review is needed
  const classes = [];

  // Parse common patterns
  if (styleObj.includes('flex: 1')) classes.push('flex-1');
  if (styleObj.includes('flexDirection: \'row\'')) classes.push('flex-row');
  if (styleObj.includes('justifyContent: \'center\'')) classes.push('justify-center');
  if (styleObj.includes('alignItems: \'center\'')) classes.push('items-center');
  if (styleObj.includes('backgroundColor: colors.background')) classes.push('bg-black');

  return classes.join(' ');
}

function migrat eFile(filePath) {
  console.log(`Processing: ${filePath}`);

  let content = fs.readFileSync(filePath, 'utf8');

  // Check if file uses StyleSheet
  if (!content.includes('StyleSheet.create')) {
    console.log(`  ✓ No StyleSheet found, skipping`);
    return false;
  }

  // Remove StyleSheet import
  content = content.replace(/,?\s*StyleSheet\s*,?/g, ',');
  content = content.replace(/from 'react-native';/, (match) => {
    // Clean up double commas
    const before = content.substring(0, content.indexOf(match));
    return before.match(/,\s*,/) ? match.replace(/,\s*,/, ',') : match;
  });

  // Remove StyleSheet.create block (we'll need to manually convert these)
  const styleSheetRegex = /const styles = StyleSheet\.create\({[\s\S]*?\n}\);/g;
  const hasStyles = styleSheetRegex.test(content);

  if (hasStyles) {
    console.log(`  ⚠ File has StyleSheet.create - requires manual conversion`);
    console.log(`     Run manual migration for this file`);
    return false;
  }

  // Write back
  fs.writeFileSync(filePath, content, 'utf8');
  console.log(`  ✓ Migrated successfully`);
  return true;
}

function processDirectory(dirPath) {
  const files = fs.readdirSync(dirPath, { withFileTypes: true });
  let migrated = 0;
  let failed = 0;

  for (const file of files) {
    const fullPath = path.join(dirPath, file.name);

    if (file.isDirectory()) {
      const result = processDirectory(fullPath);
      migrated += result.migrated;
      failed += result.failed;
    } else if (file.name.endsWith('.tsx')) {
      const success = migrateFile(fullPath);
      if (success) migrated++;
      else failed++;
    }
  }

  return { migrated, failed };
}

// Main execution
const screensDir = '/Users/olorin/Documents/olorin/olorin-media/bayit-plus/shared/screens';

console.log('Starting StyleSheet to TailwindCSS migration...\n');
console.log('='.repeat(60));

const { migrated, failed } = processDirectory(screensDir);

console.log('='.repeat(60));
console.log(`\nMigration complete!`);
console.log(`  ✓ Successfully migrated: ${migrated} files`);
console.log(`  ⚠ Requires manual review: ${failed} files`);
console.log(`\nNote: Files with complex StyleSheet.create() require manual conversion.`);
console.log(`Please review each file and convert style={styles.xyz} to className="..."`);

File: migrate_styles.py
#!/usr/bin/env python3
"""
Migrate StyleSheet.create() to TailwindCSS/NativeWind className
"""
import re
import os
import sys
from pathlib import Path

# Add backend to path to import ScriptConfig
sys.path.insert(0, str(Path(__file__).parent / "backend"))
try:
    from scripts.config.script_config import get_script_config
except ImportError:
    # Fallback if config not available
    get_script_config = None

# Style mappings
STYLE_MAP = {
    # Layout
    'container': 'flex-1 p-4',
    'headerActions': 'flex-row gap-2',
    'actionsRow': 'flex-row gap-1',

    # Buttons
    'filterButton': 'flex-row items-center px-3 py-2 bg-black/20 backdrop-blur-xl rounded-md border border-white/10',
    'exportButton': 'px-3 py-2 bg-blue-500 rounded-md',
    'addButton': 'px-3 py-2 bg-blue-500 rounded-md',
    'createButton': 'px-3 py-2 bg-blue-500 rounded-md',
    'actionButton': 'w-[30px] h-[30px] rounded-sm bg-gray-800 justify-center items-center',
    'saveButton': 'bg-purple-600 rounded-md py-3 items-center mt-4',
    'closeButton': 'px-4 py-2 bg-black/20 rounded-md border border-white/10',

    # Text
    'filterButtonText': 'text-sm text-white',
    'titleText': 'text-sm font-semibold text-white',
    'bodyText': 'text-sm text-gray-400',
    'dateText': 'text-xs text-gray-400',
    'statusText': 'text-xs font-semibold capitalize',

    # Cards
    'summaryCard': 'flex-1 bg-black/20 rounded-md border border-white/10 p-3 items-center',
    'statCard': 'flex-1 bg-black/20 rounded-md border border-white/10 p-3 items-center',

    # Modals
    'modalOverlay': 'flex-1 bg-black/50 justify-center items-center',
    'modalContent': 'w-[90%] max-w-[500px] bg-gray-900/95 rounded-lg p-4 border border-white/10',
    'modalTitle': 'text-xl font-bold text-white mb-4',
    'formGroup': 'mb-3',
    'formLabel': 'text-sm font-semibold text-white mb-1',
    'formInput': 'bg-gray-800 rounded-md border border-white/10 px-3 py-2 text-white text-base',
    'modalActions': 'flex-row justify-end gap-2',

    # Badges
    'statusBadge': 'px-2 py-0.5 rounded-sm self-start',
}

def remove_stylesheet_import(content):
    """Remove StyleSheet from imports"""
    # Remove StyleSheet from import statement
    content = re.sub(r',\s*StyleSheet', '', content)
    content = re.sub(r'StyleSheet,\s*', '', content)
    return content

def remove_stylesheet_block(content):
    """Remove the entire StyleSheet.create block"""
    # Find and remove const styles = StyleSheet.create({...});
    pattern = r'const styles = StyleSheet\.create\({[\s\S]*?\}\);'
    content = re.sub(pattern, '', content)
    return content

def convert_style_to_classname(content, style_name, class_name):
    """Convert style={styles.xyz} to className="..."""""
    # Handle various patterns
    patterns = [
        (f'style={{styles.{style_name}}}', f'className="{class_name}"'),
        (f'style={{[styles.{style_name},', f'style={{['), # Arrays - keep for dynamic
    ]

    for pattern, replacement in patterns:
        content = content.replace(pattern, replacement)

    return content

def process_file(filepath):
    """Process a single TypeScript file"""
    print(f"Processing: {filepath}")

    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Check if file uses StyleSheet
    if 'StyleSheet.create' not in content:
        print(f"  Skipping (no StyleSheet)")
        return

    # Step 1: Remove StyleSheet from imports
    content = remove_stylesheet_import(content)

    # Step 2: Convert style references to className
    for style_name, class_name in STYLE_MAP.items():
        content = convert_style_to_classname(content, style_name, class_name)

    # Step 3: Remove StyleSheet.create block
    content = remove_stylesheet_block(content)

    # Write back
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

    print(f"  ✓ Migrated")

def main():
    # Use ScriptConfig if available, otherwise fallback to git root
    if get_script_config:
        config = get_script_config()
        admin_dir = config.shared_dir / "screens" / "admin"
    else:
        # Fallback: detect project root via git
        try:
            import subprocess
            result = subprocess.run(
                ["git", "rev-parse", "--show-toplevel"],
                capture_output=True,
                text=True,
                check=True,
            )
            project_root = Path(result.stdout.strip())
            admin_dir = project_root / "shared" / "screens" / "admin"
        except:
            admin_dir = Path(__file__).parent / "shared" / "screens" / "admin"

    admin_dir = str(admin_dir)

    for filename in os.listdir(admin_dir):
        if filename.endswith('.tsx') and filename != 'index.ts':
            filepath = os.path.join(admin_dir, filename)
            process_file(filepath)

if __name__ == '__main__':
    main()
